collection_script_draft1.py

```diff
@@ -0,0 +1,409 @@
+options = webdriver.ChromeOptions()
+options.add_experimental_option("mobileEmulation", mobile_emulation)
+
+def get_ad_info(driver: webdriver.Chrome, movie_id):
+    '''
+    This function helps us get ad-data from the main video
+    '''
+    ad_id = driver.execute_script(
+        'return document.getElementsByClassName("html5-video-info-panel-content")[0].children[0].children[1].textContent.replace(" ","").split("/")[0]'
+    )
+    # sleeping for 0.5 seconds for the above function to get executed
+    time.sleep(0.5)
+
+    skippable_add: int = driver.execute_script(
+        'return document.getElementsByClassName("ytp-ad-skip-button-container").length'
+    )
+    print(f'ad is skippable? {skippable_add}')
+
+    if skippable_add:
+        try:
+
+            skip_duration = int(
+                driver.execute_script(
+                    'return document.getElementsByClassName("ytp-ad-text ytp-ad-preview-text")[0].innerText'
+                )
+            )
+        except:
+            skip_duration = -2  # Error occured, Could not get Skip Duration
+    else:
+        # sets the skip_duration to 999 meaning it was 999 seconds and could not be skipped
+        skip_duration = 999  # Add was not skippable.
+
+    # this gets the resolution of the ad when it starts
+    start_resolution: str = driver.execute_script(
+        'return document.getElementsByClassName("html5-video-info-panel-content")[0].children[2].children[1].textContent.replace(" ","").split("/")[0]'
+    )
+    start_resolution_check = start_resolution.split("@")[0]
+
+    # to keep track of the the number of times we ran the loop
+    attempt_count = 0
+    # keeps looping until we get the correct resolution of the current ad being streamed
+    while start_resolution_check == "0x0" or attempt_count < 10:
+        start_resolution = driver.execute_script(
+            'return document.getElementsByClassName("html5-video-info-panel-content")[0].children[2].children[1].textContent.replace(" ","").split("/")[0]'
+        )
+        start_resolution_check = start_resolution
+        attempt_count += 1
+    # sleeps for 0.5 seconds for the above operation to complete
+    time.sleep(0.5)
+
+    # keep looping unless the id of the current id is the same as the argument passed to the function
+    # for authentication purposes only.
+    while str(ad_id) == str(movie_id):
+        ad_id = driver.execute_script(
+            'return document.getElementsByClassName("html5-video-info-panel-content")[0].children[0].children[1].textContent.replace(" ","").split("/")[0]'
+        )
+
+    # returns the id of the add, logical value for if the ad is skippable, the duration of the ad to skip the video (up to 6 seconds long), and the start
+    # resolution of the ad
+    return ad_id, skippable_add, skip_duration, start_resolution
+
+
+def driver_code(driver: webdriver.Chrome):
+    # forcefully open youtube in mobile mode
+    driver.get('https://m.youtube.com/?persist_app=1&app=m')
```

Comment:
No need for this, Mobile emulator opens m.youtube even if original url doesnt contain m prefix